### DISCUSSION TITLE
Confusion about word vectors and tok2vec

### DISCUSSION THREAD

**Discussion by fcggamou** [explosion/spaCy#6220] [Help: Other Questions] [answered]
Hi, it would be really helpful for me if you would help me understand some underlying concepts about Spacy.

I understand some spacy models have some predefined static vectors, for example, for the Spanish models these are the vectors generated by FastText. 
I also understand that there is a tok2vec layer that generates vectors from tokens, and this is used for example as the input of the NER components of the model.

If the above is correct, then I have some questions:

- Does the NER component also use the static vectors? 
        - If yes, then where does the tok2vec layer comes into play?
        - If no, then is there any advantage on using the lg or md models if you only intend to use the model for e.g. the NER component?
- Is the tok2vec layer already trained for pretrained downloaded models, e.g. Spanish? 
- If I replace the NER component of a pretrained model, does it keep the tok2vec layer untouched i.e. with the learned weights?
- Is the tok2vec layer also trained when I train a NER model?
- Would the pretrain command help the tok2vec layer learn some domain-specific words that may be OOV? 

Thanks a lot!

**Comment by svlandeg** [comment, marked as answer]
I've added some replies to your questions on the related [SO post](https://stackoverflow.com/questions/64253599/spacy-confusion-about-word-vectors-and-tok2vec). I'll follow up there, and close this issue to keep the discussion in one place.
